BringgTrackingSDKDemo/MainViewController.m

```diff
@@ -129,20 +129,63 @@ - (IBAction)monitorOrder:(id)sender {
     }else{
         // if customer not signed in  then it is our job to provide order uuid from the partner - api
         // make sure the orderfield now hold uuid
-         NSString *orderuuid = self.orderField.text;
+        NSString *orderuuid = self.orderField.text;
+        NSString *sharedUUID;
+        NSString *orderCompound = self.orderCompoundField.text;
         
-        // if no order object we create one with the uuid we got from the partner api and the 'Created' status
-        GGOrder *order = [[GGOrder alloc] initOrderWithUUID:orderuuid atStatus:OrderStatusCreated];
-        
-        [self trackOrder:order];
+        if (orderCompound && orderCompound.length > 0) {
+            [self monitorOrderWithCompoundUUID:orderCompound];
+        }else{
+            // if no order object we create one with the uuid we got from the partner api and the 'Created' status
+            GGOrder *order = [[GGOrder alloc] initOrderWithUUID:orderuuid atStatus:OrderStatusCreated];
+            order.sharedLocationUUID = sharedUUID;
+            
+            [self trackOrder:order];
+        }
+
     }
     
     
     
     
 }
 
+- (void)monitorOrderWithCompoundUUID:(NSString *)compoundUUID{
+    
+    NSString *orderUUID;
+    NSString *sharedUUID;
+    NSError *error;
+
+    [GGBringgUtils parseOrderCompoundUUID:compoundUUID toOrderUUID:&orderUUID andSharedUUID:&sharedUUID error:&error];
+    
+    if (error) {
+        NSLog(@"cant monitor order :%@", error);
```

Comment:
don't you want to return the man error callback here so they will actually know whats wrong?